R/fitNullModel.R

```diff
@@ -17,25 +17,36 @@ setMethod("fitNullModel",
                    verbose = TRUE) {
 
               if (is.data.table(x)) x <- as.data.frame(x)
-              
-              desmat <- createDesignMatrix(x, outcome, covars, group.var)
 
+              desmat <- createDesignMatrix(x, outcome, covars, group.var)
               # if there was missing data, need to subset cov.mat
               if (!is.null(cov.mat)) {
                   .checkRownames(cov.mat, x)
                   if (nrow(desmat$X) < nrow(x)) {
                       ind <- which(rownames(x) %in% rownames(desmat$X))
                       cov.mat <- .covMatSubset(cov.mat, ind)
                   }
+                  cov.mat <-  .setCovMatNames(cov.mat)
               }
-              
-              .fitNullModel(y=desmat$y, X=desmat$X, covMatList=cov.mat,
+
+
+              out <- .fitNullModel(y=desmat$y, X=desmat$X, covMatList=cov.mat,
                             group.idx=desmat$group.idx, family=family,
                             start=start, AIREML.tol=AIREML.tol,
                             max.iter=max.iter, EM.iter=EM.iter,
                             drop.zeros=drop.zeros,
                             return.small=return.small,
                             verbose=verbose)
+              rownames(out$fit) <- rownames(desmat$y)
+
+              # Add model string elements here because we need the outcome string.
+              out$model <- .modelString(outcome, covars = covars, random = names(cov.mat),
+                                        group.var = group.var, inverse_normal = FALSE)
+              out$outcome <- .modelOutcomeString(outcome, inverse_normal = FALSE)
+              out$covars <- covars
```

Comment:
I ended up deciding to add this to the `fitNullModel` wrapper function instead of in any of the sub-functions for a couple reasons. One side effect is that the model will still include any dropped covariates. Here are my reasons:
* I had first considered adding it to `.fitNullModel`, which accepts the design matrix as an argument. In the case of categorical covariates, this design matrix has multiple indicator variables for that covariate (eg instead of `study`, it will have `studyB`, `studyC`, etc.). If you tried to use the colnames of the design matrix to get the list of covariates, you would end up with `y ~ studyB + studyC` instead of `y ~ study`.
* It would be possible to have `createDesignMatrix` return the list of covariates actually included in the model and then use that list element to create the model string. In this case, we could remove dropped covariates from the model, but it was faster to code it this way. It is also more consistent with `lm` to keep dropped covariates in the model string.

Reply:
I think it makes sense to have this here rather than in `.fitNullModel`. I still have a slight preference for dropped covariates to be removed from the model string (but could be convinced otherwise). It seems like `createDesignMatrix` could even be a good place to get that list of covariates.